src/main/java/com/wise23/chariteed/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,86 @@
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig {
+
+    @Autowired
+    private CustomLoginSuccessHandler successHandler;
+
+    @Bean
+    public UserDetailsService userDetailsService() {
+        return new UserServiceImpl();
+    }
+
+    @Bean
+    public BCryptPasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public AuthenticationManager authenticationManager(AuthenticationConfiguration authConfig) throws Exception {
+        return authConfig.getAuthenticationManager();
+    }
+
+    @Bean
+    public DaoAuthenticationProvider authenticationProvider() {
+        DaoAuthenticationProvider authProvider = new DaoAuthenticationProvider();
+
+        authProvider.setUserDetailsService(userDetailsService());
+        authProvider.setPasswordEncoder(passwordEncoder());
+
+        return authProvider;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+
+                http.authorizeRequests()
+                // URL matching for accessibility
+                .antMatchers("/", "/login", "/register").permitAll() //only for test environment
+                .antMatchers("/admin/**").hasAnyAuthority("ADMIN")
+                .anyRequest().authenticated()
+                .and()
+                // form login
+                .csrf().disable().formLogin()
+                .loginPage("/login")
+                .failureUrl("/login?error=true")
+                .successHandler(successHandler)
+                .usernameParameter("email")
+                .passwordParameter("password")
+                .and()
+                // logout
+                .logout()
+                .logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
+                .logoutSuccessUrl("/")
+                .and()
+                .exceptionHandling()
+                .accessDeniedPage("/access-denied");
+
+                http.authenticationProvider(authenticationProvider());
+                http.headers().frameOptions().sameOrigin();
+
+                return http.build();
+    }
+
+    @Bean
+    public WebSecurityCustomizer webSecurityCustomizer() {
+        return (web) -> web.ignoring().antMatchers("/images/**", "/js/**", "/webjars/**");
```

Comment:
Not sure here. Good, so that no outsiders can view pictures of the system, however the patients" data like sensitive images are stored in database, so I don"t see a particular reason to forbid those requests.

Reply:
i thought this means everyone can access these folders

Reply:
I commented it out. If we need it we can add it back in.